omiclearn/utils/ui_components.py

```diff
@@ -610,6 +610,13 @@ def _generate_additional_feature_selection_section(state, multiselect):
     with st.expander("Additional features"):
         st.markdown(ADDITIONAL_FEATURES_TEXT)
 
+        # Avoid user introducing target leakage via additional feature selection
+        remainder_features_without_target_class = state.remainder
```

Comment:
I think it is a valuable demonstration that you have the option to purposefully leak the target and show how this would change the performance. I think this would go nicely with the idea of a tool for exploration.
I would, therefore, rather have a warning label when this is selected. Feel free to go ahead with what you think fits best.